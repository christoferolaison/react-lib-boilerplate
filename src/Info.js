// @flow

import React from 'react'

type Props = {
  heading: string,
  body: string,
};

type State = {
  inverted: boolean
};

class Info extends React.Component<Props, State> {
  state = { inverted: false };

  toggleInverted = () => this.setState(({ inverted }) => ({ inverted: !inverted }));

  props: Props;

  render() {
    const color = this.state.inverted
      ? '#ffffff'
      : '#222624'
    const backgroundColor = this.state.inverted
      ? '#222624'
      : '#ffffff'
    return (
      <div // eslint-disable-line
        onClick={this.toggleInverted}
        style={{
          backgroundColor,
          color,
          transition: [ 'all', '500ms' ].join(' '),
          margin: 0,
          padding: 20,
          width: '100vw',
          height: '100vh',
          cursor: 'pointer',
        }}
      >
        <h1
          style={{
            fontSize: 24,
            letterSpacing: 0.5,
            fontFamily: 'sans-serif',
            display: 'flex',
            alignItems: 'center',
            margin: '0 0 20px 0',
          }}
        >
          {this.props.heading}
        </h1>
        <p style={{ margin: '0 0 20px 0', letterSpacing: 0.3, fontFamily: 'sans-serif' }}>
          {this.props.body}
        </p>
        <span style={{ margin: 0, fontSize: 100 }} role='img' aria-label='avocado'>ðŸ¥‘</span>
      </div>
    )
  }
}

export default Info
