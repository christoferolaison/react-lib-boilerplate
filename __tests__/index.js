// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import Info from '../src/Info'

describe('Component: Info', () => {
  it('should render as expected', () => {
    const tree = renderer.create(
      <Info heading='react-lib-boilerplate' body='react-lib-boilerplate' />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
