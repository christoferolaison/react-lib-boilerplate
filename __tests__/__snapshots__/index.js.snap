// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Info should render as expected 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "color": "#222624",
      "cursor": "pointer",
      "height": "100vh",
      "margin": 0,
      "padding": 20,
      "transition": "all 500ms",
      "width": "100vw",
    }
  }
>
  <h1
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontFamily": "sans-serif",
        "fontSize": 24,
        "letterSpacing": 0.5,
        "margin": "0 0 20px 0",
      }
    }
  >
    react-lib-boilerplate
  </h1>
  <p
    style={
      Object {
        "fontFamily": "sans-serif",
        "letterSpacing": 0.3,
        "margin": "0 0 20px 0",
      }
    }
  >
    react-lib-boilerplate
  </p>
  <span
    aria-label="avocado"
    role="img"
    style={
      Object {
        "fontSize": 100,
        "margin": 0,
      }
    }
  >
    ðŸ¥‘
  </span>
</div>
`;
